homeassistant/components/tuya/fan.py

```diff
@@ -177,7 +177,6 @@ def turn_on(
                     "value": int(self._speed.remap_value_from(percentage, 1, 100)),
                 }
             )
-            return
```

Comment:
This return prevents the execution flow from continuing when the above proper conditions are met; this is odd/wrong considering it adds a command to be sent just below. Removing this return allows the fan speed to be set by percentage during a turn_on command.

Reply:
Nice catch! 👍